core/src/main/java/com/amplifyframework/auth/result/step/AuthSignUpStep.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.result.step;
+
+public enum AuthSignUpStep {
```

Comment:
We debated this quite a bit in the naming meeting and landed on this because this is only used to indicate what the next step in a flow is, not what state the plugin is in. When someone calls a method which could start a flow, they will receive a boolean indicating if the operation is successful and a nextStep variable indicating if there is a next step or if it's done (in which case nextStep is not null but DONE).